platform/features/imagery/src/controllers/ImageryController.js

```diff
@@ -161,14 +161,18 @@ define(
          * @returns {boolean} falsy when a duplicate datum is given
          */
         ImageryController.prototype.updateHistory = function (datum) {
-            if (this.$scope.imageHistory.length === 0 ||
-                !_.isEqual(this.$scope.imageHistory.slice(-1)[0], datum)) {
-                var index = _.sortedIndex(this.$scope.imageHistory, datum, this.timeFormat.format.bind(this.timeFormat));
-                this.$scope.imageHistory.splice(index, 0, datum);
-                return true;
+            if (this.$scope.imageHistory.length !== 0) {
```

Comment:
I should have suggested this earlier, but perhaps this function would be easier to read if we separated out the logic that looks for duplicates into its own function? Right now, most of the code in the `updateHistory` function is actually looking for duplicates, not updating history. Then the `updateHistory` function could be something like -
 
```javascript
if (!historyContainsDatum(datum)) {
    var index = _.sortedIndex(this.$scope.imageHistory, datum, this.timeFormat.format.bind(this.timeFormat));
    this.$scope.imageHistory.splice(index, 0, datum);
    return true;
} else {
    return true
}
```

Reply:
On reflection, a function name of `isDuplicateOfLastImage()` is a better fit with its functionality.